### PR TITLE
add cfn function, checkstyle, cyclic dep

### PR DESCRIPTION
**Pull request by geronimo-iia**
used to retrieve cloudformation output value

### CODE REVIEW

**Comment by geronimo-iia** [comment]
@flou could you take time to this pull request
thanks

File: brume/config.py [hunk 1]
```diff
@@ -11,6 +11,33 @@
 import jinja2
 
 
+from stack import stack_outputs
+# current outputs of loaded stacks
+stackOutputsDefinition = {}
```
Comment (flou) [inline brume/config.py]: Please use snake_case for variable and methods names.

File: brume/config.py [hunk 2]
```diff
@@ -11,6 +11,33 @@
...
+
+def _check_key_exists(key, container, stack_name):
+    if not (key in container):
+        click.secho('[ERROR] No key {} variable in stack {}'.format(key, stack_name), err=True, fg='red')
+        exit(1)
+
+def cloudformation(stack_name, key, *subKeys):
```
Comment (flou) [inline brume/config.py]: Please use snake_case for variables in this function.

File: brume/config.py [hunk 3]
```diff
@@ -29,6 +56,16 @@ def is_git_repo():
 
 class Config():
 
+    @staticmethod
+    def cfn(stack_name, key, secondKey=None, thirdKey=None):
```
Comment (flou) [inline brume/config.py]: Can you provide a docstring for this method and update the README with an example usage of the `cfn` function in `brume.yaml`?

Also please use snake_case for variables in this method.

File: brume/stack.py [hunk 4]
```diff
@@ -40,17 +40,32 @@ def _make_parameters(params_list):
     return [{'ParameterKey': k, 'ParameterValue': v} for k, v in params_list.items()]
 
 
-def _outputs_for(stack):
+def _stack_walker(client, outputs, stack, collector):
     try:
-        outputs = cfn_client().describe_stacks(StackName=stack)['Stacks'][0].get('Outputs', {})
-        return {o['OutputKey']: o['OutputValue'] for o in outputs}
+        description = client.describe_stacks(StackName=stack)['Stacks'][0]
+        stackName = description['StackName']
```
Comment (flou) [inline brume/stack.py]: Please use snake_case for variables in this method.

File: brume/stack.py [hunk 5]
```diff
@@ -40,17 +40,32 @@ def _make_parameters(params_list):
     return [{'ParameterKey': k, 'ParameterValue': v} for k, v in params_list.items()]
 
 
-def _outputs_for(stack):
+def _stack_walker(client, outputs, stack, collector):
```
Comment (flou) [inline brume/stack.py]: Can you provide a docstring for this method?

File: brume/config.py [hunk 6]
```diff
@@ -11,6 +11,33 @@
 import jinja2
 
 
+from stack import stack_outputs
+# current outputs of loaded stacks
+stackOutputsDefinition = {}
+
+def _check_key_exists(key, container, stack_name):
+    if not (key in container):
```
Comment (flou) [inline brume/config.py]: You should change this condition to

```py
if key not in container:
```

File: brume/config.py [hunk 7]
```diff
@@ -11,6 +11,33 @@
...
+        click.secho('[ERROR] No key {} variable in stack {}'.format(key, stack_name), err=True, fg='red')
+        exit(1)
+
+def cloudformation(stack_name, key, *subKeys):
+    """
+    Return the value of the `key` in outputs of specified stack `stack_name`.
+
+    If `subKey` is specified, return the value of the `subKey` found in the value of the `key` in outputs of specified stack `stack_name`.
+    """
+    if not (stack_name in stackOutputsDefinition):
```
Comment (flou) [inline brume/config.py]: You should change this condition to

```py
if stack_name not in stackOutputsDefinition:
```

File: brume/stack.py (hunk 5 above)
Comment (geronimo-iia) [reply brume/stack.py]: done

**Comment by geronimo-iia** [comment]
@flou you could see here your requested change,

**Comment by geronimo-iia** [comment]
change are related to https://github.com/flou/brume/issues/13

**Comment by geronimo-iia** [comment]
@flou checkstyle are done on most of code,

**Comment by geronimo-iia** [comment]
@flou anytime to check this PR ?

**Pull request merged**